controllers/eventhandlers/grpcroute.go

```diff
@@ -0,0 +1,89 @@
+package eventhandlers
+
+import (
+	"context"
+	"github.com/golang/glog"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	gateway_api_v1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+)
+
+type enqueueRequestsForGRPCRouteEvent struct {
+	client client.Client
+}
+
+func NewEnqueueRequestGRPCRouteEvent(client client.Client) handler.EventHandler {
+	return &enqueueRequestsForGRPCRouteEvent{
+		client: client,
+	}
+}
+
+func (h *enqueueRequestsForGRPCRouteEvent) Create(e event.CreateEvent, queue workqueue.RateLimitingInterface) {
+	glog.V(6).Info("GRPCRoute create")
+
+	newGRPCRoute := e.Object.(*gateway_api_v1alpha2.GRPCRoute)
+	h.enqueueImpactedService(queue, newGRPCRoute)
+}
+
+func (h *enqueueRequestsForGRPCRouteEvent) Update(e event.UpdateEvent, queue workqueue.RateLimitingInterface) {
+	glog.V(6).Info("GRPCRoute update ")
+
+	oldGRPCRoute := e.ObjectOld.(*gateway_api_v1alpha2.GRPCRoute)
+	newGRPCRoute := e.ObjectNew.(*gateway_api_v1alpha2.GRPCRoute)
+
+	if !equality.Semantic.DeepEqual(oldGRPCRoute.Spec, newGRPCRoute.Spec) {
+		glog.V(6).Infof("--oldGRPCRoute %v \n", oldGRPCRoute.Spec)
+		glog.V(6).Infof("--newGRPCRoute %v \n", newGRPCRoute.Spec)
+		if newGRPCRoute.Status.RouteStatus.Parents != nil &&
+			newGRPCRoute.Status.RouteStatus.Parents[0].Conditions[0].LastTransitionTime != ZeroTransitionTime {
```

Comment:
is it guarantied that Parents and Conditions has size of 1 at least? if yes please attach reference. If not you need to do a check for size

Reply:
This is doing exactly what we do for HTTPRoute. According to the doc, an empty list indicates that the route has not been attached to any gateway. In the case of a GRPCRoute update, I suspect this is outside the scope of this issue.